### PR TITLE
Move master to CRD v1

### LINKED ISSUE #3875: [Bug] Use of CRD_v1beta1 API for the Strimzi CRDs
**Describe the bug**
Strimzi is using the deprecated BETA1 version of the `apiextensions / CustomResourceDefinition` API. Applying 0.20 against a 1.19 based cluster gives me:

```
namespace/kafka created
Warning: apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
...
Warning: apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
...
```

**Expected behavior**
The use of `v1` version from the CRD API

**Environment (please complete the following information):**
 - Strimzi version: .20.
 - Installation method: YAML
 - Kubernetes cluster: OCP 4.6

### LINKED ISSUE #4118: Access the CRs via the v1beta2 path
As part of https://github.com/strimzi/proposals/blob/master/009-crd-v1-roadmap.md#development we need to change the operators to access the CRs via the v1beta2 URL path.

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Bugfix

### Description

Now after 0.22 is branched, we can move master to Strimzi `v1beta2` API / CRD `apiextensions/v1` to get it ready for 0.23.

This should close #4118 and close #3875.

### Checklist

- [x] Write tests
- [x] Make sure all tests pass
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [x] Reference relevant issue(s) and close them after merging
- [x] Update CHANGELOG.md

### CODE REVIEW

File: api/src/test/resources/io/strimzi/api/kafka/model/041-Crd-kafkaconnect.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-apiVersion: apiextensions.k8s.io/v1
+apiVersion: apiextensions.k8s.io/v1beta1
```
Comment (tombentley) [inline api/src/test/resources/io/strimzi/api/kafka/model/041-Crd-kafkaconnect.yaml:1]: Why are these changing from `v1` to `v1beta1`?

Comment (scholzj) [reply api/src/test/resources/io/strimzi/api/kafka/model/041-Crd-kafkaconnect.yaml:1]: These are the resources used for tests. In the past, we had `apiextensions/v1beta1` in install and `apiextensions/v1` here. This PR turns it around => keeps `apiextensions/v1` in install and `apiextensions/v1beta1` here to have the old tests (for exampel the `api-conversion` tool tests etc.) still working. I think it is useful to keep them around and run the tests at least until the 0.23 release is out.

**Review by tombentley — APPROVED** [review]

**Review by ppatierno — APPROVED** [review]

**Pull request merged**